baseplate/events/queue.py

```diff
@@ -74,11 +87,31 @@ def set_field(self, key, value, obfuscate=False):
         if value is None or value == "":
             return
 
-        if not obfuscate:
-            self.payload[key] = value
+        if obfuscate and high_cardinality:
```

Comment:
Another API option that might be less confusing/error prone is to have a single parameter which takes an enum of "field type".  `def set_field(self, key, value, kind=NORMAL)`. Then we're explicit that the types are all mutually exclusive (instead of having some combinations of flags that can yell at the user) and we're more open to future extension if necessary.

We'd want to support `obfuscate` still as a deprecated param though, like https://github.com/reddit/baseplate/commit/6f0981bccebdf847b39211fd17016fd7a63edc58#diff-e23a2f0193db987b0509d66964675524R180